### ISSUE TITLE
AlexNet no longer able to reproduce ILSVRC 2012 results

### ISSUE THREAD

**Issue by steph-w** [BVLC/caffe#6382]
### Issue summary
Running 
`./build/tools/caffe test --model=models/bvlc_alexnet/train_val.prototxt --weights= weights/bvlc_alexnet.caffemodel --iterations=1000 --gpu 0`

Top-1: 0.568279
Top-5: 0.799501
Which is lower than the official accuracies: "top-1 accuracy 57.1% and a top-5 accuracy 80.2% on the validation set"

### Steps to reproduce
To reproduce, I downloaded 2012 ImageNet validation set using this [handy script](https://github.com/tensorflow/models/blob/master/research/inception/inception/data/download_imagenet.sh)  
Unpacked the images into an LMDB, resizing to 256 x 256 following Caffe's ImageNet [tutorial](http://caffe.berkeleyvision.org/gathered/examples/imagenet.html)  
I'm using the `imagenet_mean.binaryproto` obtained by running `get_ilsvrc_aux.sh`  
Aside from changing paths in train_val.prototxt and adding a layer to compute top-5 there are no other changes.   

Is anyone else able to reproduce this issue? 

### System configuration

* Python version (if using pycaffe): 2.7.6

**Comment by naibaf7** [comment]
It's probably important to know which CPU/GPU and which method for BLAS/Convolution have been used in each case. That difference would not surprise me when e.g. switching from an older to newer version of cuDNN or similar changes in the compute paths.
That's because float addition is not associative.

**Comment by steph-w** [comment]
@naibaf7 Wow I could not have thought of that. Here's what we're using:
GPU: Tesla M60  
CUDA release 7.5, V7.5.17  
BLAS Atlas
Thanks a lot!

**Comment by naibaf7** [comment]
@steph-w 
Try to run in on the CPU as a comparsion. Will it give you the same result?

**Comment by HelloTobe** [comment]
Same Problem as @steph-w!

Same procedure as **steph-w** , got

Top-1: 0.56818
Top-5: 0.799461

**System info**:

- GPU: Tesla K80
- CUDA: 8.0.61
- CUDNN: 5.1.10

How to check out the reason?

**Comment by naibaf7** [comment]
@HelloTobe 
What's your result using CPU instead of GPU inference?

**Comment by HelloTobe** [comment]
@naibaf7 Thanks for your attention!
The CPU inference results is as following:
Top-1: 0.56818
Top-5: 0.799461
Absolutely the same as GPU.

**Comment by AshleyRoth** [comment]
Hi! Can someone link to manual how to train caffe? Thanks
